test/unit/controllers/consolidations/AssociateDucrSummaryControllerSpec.scala

```diff
@@ -73,131 +62,102 @@ class AssociateDucrSummaryControllerSpec extends ControllerSpec with MockSubmiss
+        val result = controller.displayPage()(getRequest())
 
-      when(mockCustomsCacheService.fetchAndGetEntry[MucrOptions](any(), meq(AssociateDucr.formId))(any(), any(), any()))
-        .thenReturn(Future.successful(None))
+        status(result) mustBe OK
+        verify(mockAssociateDucrSummaryPage).apply(any(), any())(any(), any())
 
-      assertThrows[IncompleteApplication] { await(controller.displayPage()(getRequest())) }
+        val (associateDucr, mucrOptions) = theResponseData
+        associateDucr.ducr mustBe "DUCR"
+        mucrOptions mustBe "MUCR"
+      }
     }
 
-    "return Ok for GET request" in {
-
-      val result = controller.displayPage()(getRequest())
+    "retrun 500 (INTERNAL_SERVER_ERROR)" when {
 
-      status(result) mustBe OK
-    }
-  }
+      "submission service return status different than ACCEPTED" in {
 
-  "Associate DUCR Summary on POST" when {
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+        withCaching(AssociateDucr.formId, Some(AssociateDucr("DUCR")))
+        mockCustomsCacheServiceClearedSuccessfully()
+        mockDucrAssociation(BAD_REQUEST)
 
-    "MUCROptions cache is empty" should {
-      "throw incomplete application" in {
-
-        when(mockCustomsCacheService.fetchAndGetEntry[MucrOptions](any(), meq(MucrOptions.formId))(any(), any(), any()))
-          .thenReturn(Future.successful(None))
+        val result = controller.submit()(postRequest(Json.obj()))
 
-        assertThrows[IncompleteApplication] { await(controller.submit()(postRequest(Json.obj()))) }
+        status(result) mustBe INTERNAL_SERVER_ERROR
       }
     }
 
-    "AssociateDUCR cache is empty" should {
-      "throw incomplete application" in {
-
-        when(
-          mockCustomsCacheService.fetchAndGetEntry[MucrOptions](any(), meq(AssociateDucr.formId))(any(), any(), any())
-        ).thenReturn(Future.successful(None))
-
-        assertThrows[IncompleteApplication] { await(controller.submit()(postRequest(Json.obj()))) }
-      }
-    }
+    "throw an IncompleteApplication exception" when {
 
-    "everything works correctly" should {
+      "Mucr Options is missing during displaying page" in {
 
-      "return SeeOther code" in {
+        withCaching(MucrOptions.formId, None)
 
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+        assertThrows[IncompleteApplication] {
+          await(controller.displayPage()(getRequest()))
+        }
+      }
 
-        val result = controller.submit()(postRequest(Json.obj()))
+      "Associate Ducr is missing during displaying page" in {
 
-        status(result) mustBe SEE_OTHER
-      }
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
```

Comment:
Is this line necessary for test to run correctly?

Reply:
yes, because to test that DUCR is empty we need to have MUCR, in other case the method will fail on MUCR